src/ethereum_spec_tools/sync.py

```diff
@@ -499,23 +296,87 @@ def fetch_blocks_eth(
+                                    if t["to"]
+                                    else Bytes0(b""),
+                                    hex_to_u256(t["value"]),
+                                    hex_to_bytes(t["input"]),
+                                    hex_to_u256(t["v"]),
+                                    hex_to_u256(t["r"]),
+                                    hex_to_u256(t["s"]),
+                                )
                             )
-                        )
+                        else:
+                            if t["type"] == "0x1":
+                                access_list = []
+                                for sublist in t.get("accessList", []):
+                                    access_list.append(
+                                        (
+                                            self.module(
+                                                "utils.hexadecimal"
+                                            ).hex_to_address(
+                                                sublist.get("address")
+                                            ),
+                                            [
+                                                hex_to_bytes32(key)
+                                                for key in sublist.get(
+                                                    "storageKeys"
+                                                )
+                                            ],
+                                        )
+                                    )
+                                transactions.append(
+                                    b"\x01"
+                                    + rlp.encode(
+                                        self.module(
+                                            "eth_types"
+                                        ).AccessListTransaction(
+                                            Uint64(1),
+                                            hex_to_u256(t["nonce"]),
+                                            hex_to_u256(t["gasPrice"]),
+                                            hex_to_u256(t["gas"]),
+                                            self.module(
+                                                "utils.hexadecimal"
+                                            ).hex_to_address(t["to"])
+                                            if t["to"]
+                                            else Bytes0(b""),
+                                            hex_to_u256(t["value"]),
+                                            hex_to_bytes(t["input"]),
+                                            access_list,
+                                            hex_to_u256(t["v"]),
+                                            hex_to_u256(t["r"]),
+                                            hex_to_u256(t["s"]),
+                                        )
+                                    )
+                                )
```

Comment:
Can we make a few functions out of these? Not necessarily to reuse code, but to reduce the indentation level a bit :rofl: